examples/rbac/nginx/README.md

```diff
@@ -0,0 +1,138 @@
+# Role Based Access Control
+
+This example demontrates how to apply role based access control
+
+## Overview
+
+This example applies to nginx-ingress-controllers being deployed in an
+environment with RBAC enabled.
+
+Role Based Access Control is comprised of four layers:
+
+1.  `ClusterRole` - permissions assigned to a role that apply to an entire cluster
+2.  `ClusterRoleBinding` - binding a ClusterRole to a specific account
+3.  `Role` - permissions assigned to a role that apply to a specific namespace
+4.  `RoleBinding` - binding a Role to a specific account
+
+In order for RBAC to be applied to an nginx-ingress-controller, that controller
+should be assigned to a `ServiceAccount`.  That `ServiceAccount` should be
+bound to the `Role`s and `ClusterRole`s defined for the
+nginx-ingress-controller.
+
+## Service Accounts created in this example
+
+One ServiceAccount is created in this example, `nginx-ingress-serviceaccount`.
+
+## Permissions Granted in this example
+
+There are two sets of permissions defined in this example.  Cluster-wide
+permissions defined by the `ClusterRole` named `nginx-ingress-clusterrole`, and
+namespace specific permissions defined by the `Role` named
+`nginx-ingress-role`.
+
+### Cluster Permissions
+
+These permissions are granted in order for the nginx-ingress-controller to be
+able to function as an ingress across the cluster.  These permissions are
+granted to the ClusterRole named `nginx-ingress-clusterrole`
+
+* `configmaps`, `endpoints`, `nodes`, `pods`, `secrets`: list, watch
+* `services`: get, list, watch
+* `ingresses`: get, list, watch
+* `events`: create
+* `ingresses/status`: update
+
+### Namespace Permissions
+
+These permissions are granted specific to the nginx-ingress namespace.  These
+permissions are granted to the Role named `nginx-ingress-role`
+
+* `configmaps`,`secrets`,`pods`: get
+* `endpoints`: get, create, update
```

Comment:
- Could you maybe order the list alphabetically ascending, i.e. `create, get, update`?
- I had to specify `endpoints: patch` as well (https://github.com/kubernetes/ingress/issues/575#issuecomment-299192689)